MIPs/mip-mc-handshake.md

```diff
@@ -0,0 +1,255 @@
+        "accounts:" ["eip155:1:0x0910e12C68d02B561a34569E1367c9AAb42bd810"]
+      },
+      "eip155:59144":{
+        "methods": ["eth_sendTransaction", "eth_getBalance","eth_blockNumber","eth_getTransactionCount","wallet_watchAsset"],
+        "notifications": ["message"],
+        "accounts":["eip155:59144:0x0910e12C68d02B561a34569E1367c9AAb42bd810"]
+      },
+      "wallet":{
+        "methods": ["wallet_requestPermissions","wallet_getPermissions","wallet_revokePermissions", "eth_signTypedData_v4", "personal_sign", "wallet_addEthereumChain"]
+      },
+    },      
+    "sessionProperties":{
+      /* TBD */          
+    }
+  }
+}
+```
+
+
+
+# Proposal
+
+## Language
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written in uppercase in this document are to be interpreted as described in RFC 2119.
+
+## Definitions
+
+**Wallet API**: [The interface](https://docs.metamask.io/wallet/concepts/apis/) that dapps can use to programmatically interact with MetaMask.
+
+**Multichain API**: A version of the Wallet API that is optimized for interactions across multiple chains.
+
+**Scope**: A uniquely identified domain for which authorizations can be applied (see [CAIP-217](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-217.md) for further definition).
+
+**Authorization**: A permission to use a resource. This term may be used interchangeably with the term **Permission**.
+
+**eip155 scope**: A scope that applies to a particular EVM-compatible chain identified as specified in [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md).
+
+**wallet scope**: A scope that applies to capabilities that a MetaMask Wallet client can handle in a relatively self-contained way.
+
+**chain**: A distributed ledger that can be targeted. It may or may not refer to a blockchain, which is a type of distributed ledger.  
+
+**network**: An RPC network or node that enables interactions with a chain.
+
+
+## Proposal Specification
+
+### Implementing `provider_authorize`
+
+- While `provider_authorize` will not be available through an EIP-1193 provider, the Multichain API Delivery MIP provides details about how to access the multichain API. 
+
+- Authorization requests for both `wallet` and chain-specific scopes SHOULD generally be placed into `optionalScopes`. Use of `requiredScopes` will be supported but is NOT RECOMMENDED
+
+- `chainId` uniquely indentifies the target chain, formatted as a string that includes both the namespace and the reference, following CAIP-2 conventions.
+  
+- `methods`, `notifications`, and `rpcEndpoints` are listed directly under each scope in line with CAIP-25.
+
+- Additional metadata fields in `rpcEndpoints` SHOULD be compliant with the fields specified in EIP-3085.
+
+[ TODO a PR in the api-spec repo describing the provider_authorize method in OpenRPC should be added here ]
```

Comment:
please tag me in that PR whenever it's open if i can be of any service!

Reply:
ditto